src/libutil/util.c

```diff
@@ -2648,7 +2648,7 @@ rspamd_tm_to_time (const struct tm *tm, glong tz)
 	result += tm->tm_sec;
 
 	/* Now apply tz offset */
-	result -= offset;
```

Comment:
Yep. Positive time adjustments were actually subtracted, maybe negative adjustments would not have worked correctly either. Both cases seem to be handled correctly now.